test/test_datasource.py

```diff
@@ -22,6 +26,19 @@ class DataSourceFieldsTDS(unittest.TestCase):
 
     def setUp(self):
         self.ds = Datasource.from_file(TEST_TDS_FILE)
+        self.to_delete = set()
+
+    def cleanUp(self):
+        for path in self.to_delete:
+            if os.path.isdir(path):
+                shutil.rmtree(path, ignore_errors=True)
+            elif os.path.isfile(path):
+                os.unlink(path)
+
+    def get_temp_file(self, filename):
```

Comment:
We have 3 ways of doing temp file management now.

`xfile.temporary_directory` has a context manager that creates and deletes a temp dir
`bvt.py` does a less advanced version of your cleanup methods here
Then these cleanup methods here.

This can be another PR, maybe we want to unify them, at least across the tests?